### PR TITLE
Update rancher/webhook

### PR DESCRIPTION
**Pull request by cmurphy**
## Commit 1:  Allow marshalling map[any]any for chart values

Without this change, if the rancher-config config map has nested objects
as chart values, e.g.:

```
data:
  rancher-webhook: |-
    auth:
      clientCA: foo
```

The resulting marshalled YAML will be a map[any]any, and the
unmarshalled object cannot be decoded by the catalog installer. This
change uses an alternative library so that nested values can be encoded.

## Commit 2 Update rancher/webhook

Depends on https://github.com/rancher/webhook/pull/273 and upcoming charts change

## Problem

The webhook has new optional chart values that need to be configurable by the user. These values are nested, and the current implementation for configuring webhook chart values does not support nested objects.
 
## Solution

Use an alternate library to encode charts values and bump the webhook version.
 
### Manual Testing

- built a custom charts repo with the updated webhook change and configured rancher to use it instead of git.rancherio
- see linked issue for full configuration

### Automated Testing

* Test types added/modified:
    * Unit

Summary: _TODO_

### Regressions Considerations

With no new configmap values set, webhook should continue to deploy on rancher server and agents and work as normal.

Existing / newly added automated tests that provide evidence there are no regressions:
* catalogv2 unit tests

### CODE REVIEW

**Comment by cmurphy** [comment]
@MbolotSuse pointed out that there is already a mechanism for configuring custom values.yaml for the webhook, but I found it did not fully work when values.yaml included nested objects, so I pivoted this PR to fix that.

**Review by KevinJoiner — COMMENTED** [review]
LGTM after the charts are updated

File: pkg/catalogv2/system/system_test.go [hunk 1]
```diff
@@ -269,7 +288,11 @@ func TestIsInstalled(t *testing.T) {
 			assert.Equal(t, test.expectedInstalled, installed)
 			assert.Equal(t, test.expectedVersion, version)
 			assert.Equal(t, test.expectedValues, values)
-			assert.Equal(t, test.expectedErr, err != nil)
+			if test.expectedErr {
```
Comment (maxsokolovsky) [inline pkg/catalogv2/system/system_test.go:291]: Could you add the same adjustment to the other test - TestIsInstalledExactVersion?

Comment (cmurphy) [reply pkg/catalogv2/system/system_test.go:291]: done

**Comment by maxsokolovsky** [comment]
I wonder if, in general, we should adjust the YAML parser and not JSON?

There has just been a PR that left JSON as is, but used a different YAML parser.
https://github.com/rancher/rancher/pull/42345

**Comment by cmurphy** [comment]
@maxsokolovsky thanks, with #42345 merged the problem is now fixed on the YAML end so we don't need to change the decoding on the JSON end. This PR is now just about bumping the webhook version.

**Comment by cmurphy** [comment]
Closing in favor of https://github.com/rancher/rancher/pull/42306, this change no longer has any extra functionality besides updating the webhook version

**Pull request closed without merging**